plugins/httpapi/mso.py

```diff
@@ -187,6 +160,28 @@ def send_request(self, method, path, data=None):
             raise ConnectionError(json.dumps(self._verify_response(None, method, full_path, None)))
         return self._verify_response(response, method, full_path, rdata)
 
+    def set_connection_parameters(self):
+        connection_parameters = {}
+        for key in CONNECTION_KEYS:
+            if key == "login_domain":
+                value = self.params.get(key) if self.params.get(key) is not None else self.get_option(CONNECTION_MAP.get(key, key))
+                connection_parameters[key] = value
+                self.set_option(key, value)
+            else:
+                value = self.params.get(key) if self.params.get(key) is not None else self.connection.get_option(CONNECTION_MAP.get(key, key))
+                connection_parameters[key] = value
+                self.connection.set_option(key, value)
+
+            if value != self.connection_parameters.get(key) and key in RESET_KEYS:
+                self.connection._connected = False
```

Comment:
Does this clear _auth too?

Reply:
Also why doing the reset here and doing the login in another if condition?

Reply:
> Does this clear _auth too?

This does not clear _auth because it is not needed to clear when `_connected = False`.  The send method of the connection class has a decorator [@ensure_connect](https://github.com/ansible-collections/ansible.netcommon/blob/main/plugins/connection/httpapi.py#L276). 

The [decorator function](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/__init__.py#L35) calls the [connect](https://github.com/ansible-collections/ansible.netcommon/blob/main/plugins/connection/httpapi.py#L247) method which calls our login method (which on success) overwrites the new Bearer token to _auth.

I can clear _auth but don't think it is needed in our current flow.

Reply:
> Also why doing the reset here and doing the login in another if condition?

I am not sure what you mean with this question, could you elaborate a bit more.